product-web-app/service/authService.ts

```diff
@@ -0,0 +1,30 @@
+import axios from 'axios';
+import User from '../model/user.js';
+import AuthValidator from './authValidator.js';
+
+export default class AuthService {
+  private authValidator: AuthValidator;
+
+  constructor(authValidator: AuthValidator) {
+    this.authValidator = authValidator;
+  }
+
+  async register(user: User): Promise<void> {
+    const validatePasswordError = this.authValidator.validatePassword(user.password);
+    if (validatePasswordError) {
+      throw new Error(validatePasswordError);
+    }
+
+    const validateEmailError = this.authValidator.validateEmail(user.email);
+    if (validateEmailError) {
+      throw new Error(validateEmailError);
+    }
+
+    try {
+      const response = await axios.post('http://localhost:8080/api/auth/register', user);
```

Comment:
remove the base url, setup it in app.ts see in example project or in other PR's
https://github.com/ljercha/agile-project/blob/fd328a0b0e3de1bc7714810cf5d2c308a447f879/product-web-app/app.ts#L32C34-L32C34